content/en/tracing/profiler/intro_to_profiling.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Intro to Profiling
+kind: Documentation
+---
+
+You may have heard that profiling can make your services faster, cheaper, and more reliable, but if you haven't used a profiler before, it can seem like a dark art.
+
+In this guide, we'll explain profiling, then profile a sample service that has a performance problem and see how we can use Continuous Profiler to understand and fix the problem!
+
+## What is Profiling?
+
+A profiler shows how much "work" each line of code is doing by collecting data about the program as it's running. For example, infrastructure monitoring might show that your app servers are using 80% of their CPU but you may not know why. Profiling could then show you that function `doSomeWork` is using 48% of the CPU, function `renderGraph` is using another 19%, and so on. This is important when working on performance problems because many programs spend a lot of time in very few places and it's often not obvious which places those are. Guessing at which parts of a program to optimize can often cause engineers to spend a lot of time and not get much in the way of results. By using a profiler, we find exactly which parts of the code we should optimize to get the most bang for our buck.
```

Comment:
NIT: On first reading the phrase `"work" each line of code is doing` felt confusing. Especially since the next sentences focus on functions not lines. Also the explanation of what "work" means is 2 paragraphs away, so the reader might feel like this is a "dark art" after all until he gets to this point.

Not entirely sure what the best way to improve is. Maybe using "function" consistently instead of "line" would help simplify things a bit. Or just removing the first sentence and adjusting the second one a bit.

Reply:
Good call on the mismatch between loc and function. I'll change to function everywhere.